tests/remotesecret/current-image-pull-remote-secret.go

```diff
@@ -0,0 +1,150 @@
+			// Initialize the tests controllers
+			fw, err = framework.NewFramework(utils.GetGeneratedNamespace("spi-demos"))
+			Expect(err).NotTo(HaveOccurred())
+			namespace = fw.UserNamespace
+			Expect(namespace).NotTo(BeEmpty())
+		})
+
+		AfterAll(func() {
+			if !CurrentSpecReport().Failed() {
+				Expect(fw.SandboxController.DeleteUserSignup(fw.UserName)).To(BeTrue())
+			}
+		})
+
+		It("creates an application", func() {
+			createdApplication, err := fw.AsKubeDeveloper.HasController.CreateApplication(applicationName, namespace)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(createdApplication.Spec.DisplayName).To(Equal(applicationName))
+			Expect(createdApplication.Namespace).To(Equal(namespace))
+		})
+
+		It("checks if application is healthy", func() {
+			Eventually(func() string {
+				appstudioApp, err := fw.AsKubeDeveloper.HasController.GetApplication(applicationName, namespace)
+				Expect(err).NotTo(HaveOccurred())
+				application = appstudioApp
+
+				return application.Status.Devfile
+			}, 3*time.Minute, 100*time.Millisecond).Should(Not(BeEmpty()), fmt.Sprintf("timed out waiting for gitOps repository to be created for the %s application in %s namespace", applicationName, fw.UserNamespace))
+
+			Eventually(func() bool {
+				gitOpsRepository := utils.ObtainGitOpsRepositoryName(application.Status.Devfile)
+
+				return fw.AsKubeDeveloper.CommonController.Github.CheckIfRepositoryExist(gitOpsRepository)
+			}, 1*time.Minute, 1*time.Second).Should(BeTrue(), fmt.Sprintf("timed out waiting for HAS controller to create gitops repository for the %s application in %s namespace", applicationName, fw.UserNamespace))
+		})
+
+		It("creates an environment", func() {
+			_, err = fw.AsKubeDeveloper.GitOpsController.CreatePocEnvironment(environmentName, namespace)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("creates component detection query", func() {
+			cdq, err = fw.AsKubeDeveloper.HasController.CreateComponentDetectionQuery(applicationName, namespace, gitSourceUrl, "", "", secret, false)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("creates component", func() {
+			for _, compDetected := range cdq.Status.ComponentDetected {
+				c, err := fw.AsKubeDeveloper.HasController.CreateComponent(compDetected.ComponentStub, namespace, "", secret, applicationName, true, map[string]string{})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(c.Name).To(Equal(compDetected.ComponentStub.ComponentName))
+
+				componentList = append(componentList, c)
+			}
+
+			Expect(componentList).To(HaveLen(1))
+			component = componentList[0]
+		})
+
+		It("checks if image pull remote secret was deployed", func() {
+			Eventually(func() bool {
+				imagePullRemoteSecret, err = fw.AsKubeAdmin.RemoteSecretController.GetImageRemoteSecret("-pull", applicationName, component.Spec.ComponentName, namespace)
+				Expect(err).NotTo(HaveOccurred())
+
+				return meta.IsStatusConditionTrue(imagePullRemoteSecret.Status.Conditions, "Deployed")
```

Comment:
Do we need to check `reason: Injected` as well? Or `type: Deployed` guarantees the condition?

Reply:
From what I know, `type: Deployed` guarantees `reason: Injected`

Reply:
More info: https://github.com/redhat-appstudio/remote-secret/blob/main/controllers/remotesecret_controller.go#L372-L373